modules/library/main/src/main/java/org/geotools/renderer/crs/ProjectionHandler.java

```diff
@@ -714,18 +714,23 @@ public Geometry preProcess(Geometry geometry) throws TransformException, Factory
             mask = JTS.transform(maskWgs84, CRS.findMathTransform(WGS84, geometryCRS));
         }
 
-        return densify(intersect(geometry, mask, geometryCRS));
+        return densify(intersect(geometry, mask, geometryCRS), false);
     }
 
     /**
      * Densifies the given geometry using the current densification configuration.
      *
      * <p>It returns the original geometry if densification is not enabled.
      */
-    protected Geometry densify(Geometry geometry) {
+    protected Geometry densify(Geometry geometry, boolean validate) {
         if (geometry != null && densify > 0.0) {
             try {
-                geometry = Densifier.densify(geometry, densify);
+                // disable validation, it runs an expensive buffer operation that
+                // can bring the VM to an OOM when run on large geometries.
+                Densifier densifier = new Densifier(geometry);
+                densifier.setDistanceTolerance(densify);
+                densifier.setValidate(false);
```

Comment:
shouldn't you use the validate parameter instead of constant "false"?

Reply:
Good catch.